src/Nethermind/Nethermind.JsonRpc.Test/JsonRpcSocketsClientTests.cs

```diff
@@ -81,7 +81,7 @@ static async Task<int> CountNumberOfMessages(Socket socket, CancellationToken to
                     while (true)
                     {
                         ReceiveResult? result = await stream.ReceiveAsync(buffer).ConfigureAwait(false);
-                        if (result?.EndOfMessage == true)
+                        if (result is not null && result.EndOfMessage && (!result.Closed || result.Read != 0))
```

Comment:
Should we make `result.EndOfMessage && (!result.Closed || result.Read != 0)` into extension method I see it used 2 times

Reply:
Moved to a method. As it's used only these tests, I don't wanna make it an extension method so as not to clutter the `ReceiveResult` interface.